wandb/sdk/internal/sender.py

```diff
@@ -890,6 +892,32 @@ def _setup_fork(self, server_run: dict):
         self._run.forked = True
         self._run.starting_step = first_step
 
+    def _load_rewind_state(self, run: "RunRecord"):
+        assert self._settings.resume_from
+        self._rewind_response = self._api.rewind_run(
+            run_name=run.run_id,
+            entity=run.entity or None,
+            project=run.project or None,
+            metric_name=self._settings.resume_from.metric,
+            metric_value=self._settings.resume_from.value,
+            program_path=self._settings.program or None,
+        )
+        self._resume_state.history = self._rewind_response.get("historyLineCount", 0)
+        self._resume_state.config = json.loads(
+            self._rewind_response.get("config", "{}")
+        )
+
+    def _setup_rewind(self):
+        assert self._settings.resume_from
+        assert self._settings.resume_from.metric == "_step"
+        assert self._run
+        assert self._rewind_response
+
+        first_step = int(self._settings.resume_from.value) + 1
```

Comment:
what if the step is 0 and we didn't log anything does it mean we will have an offset by 1?

Reply:
yeah this is a good question. i guess in the current formulation the step is exclusive. we could make it inclusive - would that be more intuitive?

Reply:
i think it makes sense to be exclusive for fork, but maybe it should be inclusive for rewind

Reply:
made this inclusive (i.e., the situation you describe above will no longer happen)

Reply:
on second thought I think this needs to match the behavior in fork, because the keysInfo and history recomputation on the backend assume exclusivity here. But I think the example provided is a bit contrived. There is not much use rewinding to zero, that is effectively the same as starting a new run. So there is a clear workaround in that case - just call `wandb.init()` normally